runtime/src/main/java/org/corfudb/runtime/view/replication/ChainReplicationProtocol.java

```diff
@@ -61,6 +66,42 @@ public ILogData peek(Layout layout, long globalAddress) {
         return ret == null || ret.isEmpty() ? null : ret;
     }
 
+    /**
+     * Read data from a range created from the given addresses.
+     *
+     * This will be changed when an API is added to support multiple of ranges
+     *
+     * @param layout                The layout to use for the readAll.
+     * @param globalAddresses       A set of addresses to read from.
+     * @return                      A map of addresses to committed
+     *                              addresses, hole filling if necessary.
+     */
+    @Override
+    public Map<Long, ILogData> readAll(Layout layout, Set<Long> globalAddresses) {
+        Range<Long> range = Range.encloseAll(globalAddresses);
+        long startAddress = range.lowerEndpoint();
+        long endAddress = range.upperEndpoint();
+        int numUnits = layout.getSegmentLength(startAddress);
+        log.trace("ReadAll[{}-{}]: chain {}/{}", startAddress, endAddress, numUnits, numUnits);
```

Comment:
Unit tests don't have to have mocks. You need unit tests, since this is a bulk read on the address space, maybe you want to write your tests in AddressSpaceViewTest.